### PR TITLE
OED-383: Unit test for dmaasbackup controller

### PR DESCRIPTION
**Pull request by mynktl**
Changes:
- added test for basic controller
- added test for dmaasbackup controller

TODO:
- test for pkg/dmaasbackup

### CODE REVIEW

**Review by vishnuitta — COMMENTED** [review]
thanks @mynktl for having all these cases..
Can we have one for the 'deletionTimestamp' check that is added in shouldProcessBackup? this is something I thought might have come as bug if missed.

**Comment by vishnuitta** [comment]
so, the case can be - Set deletionTimestamp, have status as "" in the starting, and after the call, expected status should be inProgress.. (or) in some other way more precise to it

File: pkg/controller/dmaas_backup_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+			updated, err := client.MayadataV1alpha1().
+				DMaaSBackups(test.dmaasbackup.Namespace).Get(test.dmaasbackup.Name, metav1.GetOptions{})
+			require.NoError(t, err, "failed to get dmaasbackup err=%v", err)
+
+			// verify reason
+			if test.deleteErrorMsg != "" {
+				assert.Equal(t, test.deleteErrorMsg, updated.Status.Reason)
+			}
+			if test.shouldRemoveFinalizer {
```
Comment (vishnuitta) [inline pkg/controller/dmaas_backup_controller_test.go:247]: its better to have 'else' also

**Comment by mynktl** [comment]
>Can we have one for the 'deletionTimestamp' check that is added in shouldProcessBackup? this is something I thought might have come as bug if missed.
>so, the case can be - Set deletionTimestamp, have status as "" in the starting, and after the call, expected status should be inProgress.. (or) in some other way more precise to it

We have one scenario which verify `deletionTimestamp` check in shouldProcessBackup. https://github.com/mayadata-io/dmaas-operator/blob/61776c366536c664f1184b5d8774eeb69b659e23/pkg/controller/dmaas_backup_controller_test.go#L188-L193

This will ensure that controller has processed dmaasbackup object.

**Review by vishnuitta — APPROVED** [review]
changes are good

**Pull request merged**